pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -2294,17 +2295,16 @@ private CompletableFuture<Producer<byte[]>> initDeadLetterProducerIfNeeded() {
                 p = deadLetterProducer;
                 if (p == null || p.isCompletedExceptionally()) {
                     p = createProducerWithBackOff(() -> {
-                        CompletableFuture<Producer<byte[]>> newProducer =
-                                ((ProducerBuilderImpl<byte[]>) client.newProducer(Schema.AUTO_PRODUCE_BYTES(schema)))
-                                        .initialSubscriptionName(this.deadLetterPolicy.getInitialSubscriptionName())
-                                        .topic(this.deadLetterPolicy.getDeadLetterTopic())
-                                        .producerName(
-                                                String.format("%s-%s-%s-%s-DLQ", this.topicName, this.subscription,
-                                                        this.consumerName, RandomStringUtils.randomAlphanumeric(5)))
-                                        .blockIfQueueFull(false)
-                                        .enableBatching(false)
-                                        .enableChunking(true)
-                                        .createAsync();
+                        ProducerBuilderImpl<byte[]> builder = (ProducerBuilderImpl<byte[]>) client
+                                .newProducer(Schema.AUTO_PRODUCE_BYTES(schema))
+                                .topic(deadLetterPolicy.getDeadLetterTopic());
+                        builder.initialSubscriptionName(deadLetterPolicy.getInitialSubscriptionName());
+                        if (deadLetterPolicy.getDeadLetterProducerBuilderCustomizer() != null) {
+                            ProducerBuilderContext context = new ProducerBuilderContextImpl(
+                                    deadLetterPolicy.getDeadLetterTopic(), topic, subscription);
+                            deadLetterPolicy.getDeadLetterProducerBuilderCustomizer().customize(context, builder);
```

Comment:
only create the context instance and call the customizer if the customizer is non-null